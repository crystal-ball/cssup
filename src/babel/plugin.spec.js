const babel = require('babel-core')
const { readFileSync } = require('fs')
const { resolve } = require('path')

const plugin = require('./plugin')

// Runs relative to repo root
const content = readFileSync(resolve(__dirname, '../../test/content.js'))

it('transforms cssup tagged template literals', () => {
  const { code } = babel.transform(content, { plugins: [plugin] })
  expect(code).toMatchSnapshot()
})

// ⚠️ TODO: attach test fn to secret global bridge and test passed css with a snapshot!
