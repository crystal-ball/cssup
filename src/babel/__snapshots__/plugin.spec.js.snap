// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms cssup tagged template literals 1`] = `
"import cssup from 'cssup';

const regularVariable = 'test';

const staticTest = 'test--b37f78';

const testIgnoreTemplateExpression = cssip\`this template string should remain untouched\`;

const dynamicTest = ({ color, outline, size }) => cssup('btn--1cbe6d', {
  outline: outline,
  'test-string-key': size && color,
  [\`test-template-key-\${size}\`]: size && !col
});"
`;
